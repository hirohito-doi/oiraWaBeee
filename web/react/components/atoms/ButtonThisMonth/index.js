import React from 'react';
import { makeStyles } from '@material-ui/styles';
import { Button, Typography } from '@material-ui/core';
import moment from 'moment';

const useStyles = makeStyles(theme => ({
  buttonThisMonth: {
    marginRight: 15
  }
}));

const ButtonThisMonth = props => {
  const { changeMonth } = props;
  const classes = useStyles();

  const now = moment();
  const thisYear = now.year();
  const thisMonth = now.month()

  return (
    <Button
        className={classes.buttonThisMonth}
        variant="outlined"
        color="primary"
        onClick={()=>{changeMonth(thisYear, thisMonth)}}
      >
        <Typography>
          今月
        </Typography>
      </Button>
  )
}

export default ButtonThisMonth;