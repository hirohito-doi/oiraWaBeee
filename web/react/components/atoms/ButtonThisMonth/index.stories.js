import React from 'react';
import { action } from '@storybook/addon-actions';

import ButtonThisMonth from '.'

export default {
  component: ButtonThisMonth,
  title: 'ButtonThisMonth',
};

export const regular = () => (
  <ButtonThisMonth
    changeMonth={action('clicked')}
  />
);