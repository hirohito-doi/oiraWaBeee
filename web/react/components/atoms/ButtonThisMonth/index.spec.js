import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import ButtonThisMonth from '.';

describe('ButtonThisMonth', () => {
  it('render correctory', () => {
    const tree = renderer
      .create(<ButtonThisMonth changeMonth={jest.fn()} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('check click event', () => {
    const onClick = jest.fn();
    const tree = mount(<ButtonThisMonth changeMonth={onClick} />)
    
    // クリックテスト
    tree.find('button').simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
})