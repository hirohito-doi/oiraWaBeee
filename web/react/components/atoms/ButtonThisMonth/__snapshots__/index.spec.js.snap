// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonThisMonth render correctory 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-buttonThisMonth-1 MuiButton-outlined MuiButton-outlinedPrimary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      今月
    </p>
  </span>
</button>
`;
